src/hinaura/main.ts

```diff
@@ -0,0 +1,90 @@
+    let modalites_accompagnement: string = '';
+    let formatAdresse: string = '';
+
+    const hinauraLieuMediationNumerique: HinauraLieuMediationNumerique = JSON.parse(JSON.stringify(item));
+
+    for (const [key, value] of Object.entries(item)) {
+      let newKey: string = objectKeyFormatter(key);
+      if (newKey === 'ville') newKey = 'commune';
+      if (newKey === 'adresse_postale') newKey = 'adresse';
+      if (newKey === 'adresse') formatAdresse = String(value).includes('\n') ? value.substring(0, value.indexOf('\n')) : value;
+      if (newKey === 'bf_latitude') newKey = 'latitude';
+      if (newKey === 'bf_longitude') newKey = 'longitude';
+      if (newKey === 'datetime_latest') {
+        newKey = 'date_maj';
+        dateMajFormat = value.split('/');
+        dateMajFormat = `${dateMajFormat[2]?.split(' ')[0]}-${dateMajFormat[1]}-${dateMajFormat[0]}`;
+      }
+      if (newKey === 'code_postal') codePostalToString = value.toString();
+      if (newKey === 'adresse_postale') newKey = 'adresse';
+      if (newKey === 'nom_du_lieu_ou_de_la_structure') newKey = 'nom';
+
+      if (newKey === 'tarifs') conditions_access = processConditionsAccess(value);
+      if (newKey === "types_d'accompagnement_proposes") modalites_accompagnement = processModalitesAccompagnement(value);
+
+      item['id' as keyof typeof item] = index.toString();
+      item['conditions_access' as keyof typeof item] = conditions_access;
+      item['modalites_accompagnement' as keyof typeof item] = modalites_accompagnement;
+      item['publics_accueillis' as keyof typeof item] = formatPublicAccueilliField(hinauraLieuMediationNumerique).join(',');
+      item['services' as keyof typeof item] = formatServicesField(hinauraLieuMediationNumerique, modalites_accompagnement).join(
+        ', '
+      );
+      item['code_postal' as keyof typeof item] = codePostalToString;
+      item['date_maj' as keyof typeof item] = dateMajFormat;
+      item['adresse' as keyof typeof item] = formatAdresse;
+      item[newKey as keyof typeof item] = item[key as keyof typeof item];
+      delete item[key as keyof typeof item];
+      if (newKey === 'site_web') {
+        const siteWebRegex: RegExp = /^(ftp|http|https):\/\/[^ "]+$/u;
+        if (!siteWebRegex.test(value)) delete item['site_web' as keyof typeof item];
+      }
+      if (newKey === 'telephone') {
+        if (value === '') delete item['telephone' as keyof typeof item];
+      }
+    }
+  });
+
+  // we print formated data in a json but we also can use directly formatedData here
+  const dataToStringify: string = JSON.stringify(
+    hinauraLieuxMediationNumerique.filter(
+      (item: { key: string; value: any }): boolean => item['services' as keyof typeof item] !== ''
+    )
+  );
+  fs.writeFile('./assets/output/hinaura-formated.json', dataToStringify, 'utf8', (err: ErrnoException | null): void => {
+    if (err != null) {
+      console.log(err);
+      return;
+    }
+  });
```

Comment:
```suggestion
  const hinauraLieuxMediationNumerique: HinauraLieuMediationNumerique[] = JSON.parse(dataString);
  const lieuxDeMediationNumerique: LieuMediationNumerique[] = [];

  hinauraLieuxMediationNumerique.forEach(
    (hinauraLieuMediationNumerique: HinauraLieuMediationNumerique, index: number): void => {
      lieuxDeMediationNumerique.push({
        id: index.toString(),
        nom: hinauraLieuMediationNumerique['Nom du lieu ou de la structure *'],
        pivot: Pivot('00000000000000'),
        adresse: {
          code_postal: hinauraLieuMediationNumerique['Code postal'],
          commune: hinauraLieuMediationNumerique['Ville *'],
          voie: String(hinauraLieuMediationNumerique['Adresse postale *']).includes('\n')
            ? hinauraLieuMediationNumerique['Adresse postale *'].substring(
                0,
                hinauraLieuMediationNumerique['Adresse postale *'].indexOf('\n')
              )
            : hinauraLieuMediationNumerique['Adresse postale *']
        } as Adresse,
        localisation: Localisation({
          latitude: hinauraLieuMediationNumerique.bf_latitude,
          longitude: hinauraLieuMediationNumerique.bf_longitude
        }),
        contact: {
          telephone: hinauraLieuMediationNumerique.Téléphone,
          site_web: [hinauraLieuMediationNumerique['Site Web'] as Url],
          courriel:
            hinauraLieuMediationNumerique[
              "Email (éviter les emails nominatifs - en cas d'email nominitatif seule la personne concernée est autorisé à l'ajouter)"
            ]
        } as Contact,
        conditions_access: processConditionsAccess(hinauraLieuMediationNumerique.Tarifs).split(
          ','
        ) as unknown as ConditionsAccess,
        modalites_accompagnement: processModalitesAccompagnement(
          hinauraLieuMediationNumerique["Types d'accompagnement proposés"]
        ).split(',') as unknown as ModalitesAccompagnement,
        date_maj: new Date(
          `${hinauraLieuMediationNumerique.datetime_latest.split('/')[2]?.split(' ')[0]}-${
            hinauraLieuMediationNumerique.datetime_latest.split('/')[1]
          }-${hinauraLieuMediationNumerique.datetime_latest.split('/')[0]}`
        ),

        publics_accueillis: PublicsAccueillis(formatPublicAccueilliField(hinauraLieuMediationNumerique)),
        services: formatServicesField(
          hinauraLieuMediationNumerique,
          processModalitesAccompagnement(hinauraLieuMediationNumerique["Types d'accompagnement proposés"])
        ) as Services,
        source: 'Hinaura'
      });
    }
  );

  // we print formated data in a json but we also can use directly formatedData here
  const schemaLieuxDeMediationNumeriqueBlob: string = JSON.stringify(
    toSchemaLieuxDeMediationNumerique(
      lieuxDeMediationNumerique.filter(
        (lieuDeMediationNumerique: LieuMediationNumerique): boolean => lieuDeMediationNumerique.services.length > 0
      )
    )
  );
  fs.writeFile(
    './assets/output/hinaura-formated.json',
    schemaLieuxDeMediationNumeriqueBlob,
    'utf8',
    (err: ErrnoException | null): void => {
      if (err != null) {
        console.log(err);
        return;
      }
    }
  );
```

Le but est d'enlever quelques `if` dans le fichier `main` également, pour cela on supprime la boucle for qui passe en revue tous les champs provenant d'Hinaura, à la place on ajoute des lieux de médiation numérique dans un tableau en appelant directement les fonctions de transformations pour chaque champ.

J'ai mis beaucoup de `as unknown as <Model>` au lieu d'appeler directement les  fonctions Model(), cela permet de contourner les erreurs de format qui sont vérifiées par les fonctions des models dans un premier temps, par la suite on commencera à les appeler avec des fonctions de formatage pour s'assurer que les données sont parfaitement conformes à ce qui est défini dans le schéma